File: 2025/1.ts
import { fetchInput } from "./util";

async function part1() {
  const input = await fetchInput(1);
  const lines = input
    .replace(/R/g, "")
    .replace(/L/g, "-")
    .split("\n")
    .map(Number);

  let d = 50;
  let ans = 0;
  for (const rot of lines) {
    d += rot;
    d %= 100;
    if (d == 0) {
      ans++;
    }
  }

  console.log(ans);
}

async function part2() {
  const input = await fetchInput(1);
  const lines = input
    .replace(/R/g, "")
    .replace(/L/g, "-")
    .split("\n")
    .map(Number);

  let d = 50;
  let ans = 0;
  for (const rot of lines) {
    for (let i = 0; i < Math.abs(rot); i++) {
      d += rot < 0 ? -1 : 1;
      d %= 100;
      if (d == 0) {
        ans++;
      }
    }
  }

  console.log(ans);
}

part1();
part2();

File: 2025/1.py
import util


def part1(input: str) -> int:
    ans = 0
    d = 50
    for line in input.splitlines():
        dir, rot = line[0], int(line[1:])
        if dir == "L":
            d -= rot
        if dir == "R":
            d += rot

        d %= 100
        if d == 0:
            ans += 1

    return ans


def part2(input: str) -> int:
    ans = 0
    d = 50
    for line in input.splitlines():
        dir, rot = line[0], int(line[1:])
        for _ in range(rot):
            if dir == "L":
                d -= 1
            if dir == "R":
                d += 1

            d %= 100

            if d == 0:
                ans += 1

    return ans


if __name__ == "__main__":
    util.run(1, part1, part2)
